src/Main.elm

```diff
@@ -1,10 +1,362 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Audience as Audience
+import Data.AudienceFolder as AudienceFolder
+import Decoder
+import Html exposing (Html, button, div, text, ul)
+import Html.Attributes exposing (class)
+import Html.Events exposing (onClick)
+import Http exposing (Error(..))
 
 
-main : Html msg
+type Msg
+    = MsgGotResultsAudienceFolder (Result Http.Error (List AudienceFolder.AudienceFolder))
+    | MsgGotResultsAudience (Result Http.Error (List Audience.Audience))
+    | OpenFolder Int
+    | GoBack
```

Comment:
Usually it's best to pass some sort of data to avoid `Msg` duplication (the DOM is your enemy). So for example here I'd use something like `GoToParentFolder (Maybe Int)` for this purpose, but it's not a big deal.